octree_web_viewer/src/backend.rs

```diff
@@ -65,82 +68,98 @@ fn pad(input: &mut Vec<u8>) {
+                // an Array with is very slow.
+                // The alternative is to binary encode the whole request and parse it on the client side,
+                // which requires careful constructing on the server and parsing on the client.
+                let mut reply_blob = Vec::<u8>::new();
+
+                let mut num_nodes_fetched = 0;
+                let mut num_points = 0;
+                for node_id in nodes_to_load {
+                    let mut node_data = match octree.get_node_data(&node_id) {
+                        Ok(n) => n,
+                        Err(err) => panic!("Could not get node {}: {} ", node_id, err),
+                    };
+
+                    // Write the bounding box information.
+                    let min = node_data.meta.bounding_cube.min();
+                    reply_blob.write_f32::<LittleEndian>(min.x).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.y).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.z).unwrap();
+                    reply_blob
+                        .write_f32::<LittleEndian>(node_data.meta.bounding_cube.edge_length())
+                        .unwrap();
+
+                    // Number of points.
+                    reply_blob
+                        .write_u32::<LittleEndian>(node_data.meta.num_points as u32)
+                        .unwrap();
+
+                    // Position encoding.
+                    let bytes_per_coordinate =
+                        node_data.meta.position_encoding.bytes_per_coordinate();
+                    reply_blob.write_u8(bytes_per_coordinate as u8).unwrap();
+                    assert!(
+                        bytes_per_coordinate * node_data.meta.num_points as usize * 3
+                            == node_data.position.len()
+                    );
+                    assert!(node_data.meta.num_points as usize * 3 == node_data.color.len());
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.position);
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.color);
+                    pad(&mut reply_blob);
+
+                    num_nodes_fetched += 1;
+                    num_points += node_data.meta.num_points;
+                }
+
+                let duration_ms = (time::precise_time_ns() - start) as f32 / 1000000.;
+                println!(
+                    "Got {} nodes with {} points ({}ms).",
+                    num_nodes_fetched, num_points, duration_ms
+                );
+
+                result(Ok(HttpResponse::Ok()
+                    .content_type("application/octet-stream")
+                    //disabling default encoding: 2x speed obtained while sending 10% more data
+                    .content_encoding(ContentEncoding::Identity)
+                    .body(reply_blob)))
+            }) // Construct boxed future by using `AsyncResponder::responder()` method
```

Comment:
I do not understand this comment - can you make it a full sentence?

Reply:
done. i had to research a lot for the syntax of the responder, so i put some text for future readers